pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateEvaluator.java

```diff
@@ -105,6 +105,15 @@ default int applySV(int limit, int[] docIds, int[] values) {
    * APIs for dictionary based predicate evaluator
    */
 
+  /**
+   * return the number of matching items specified by predicate
+   * negative number indicates exclusive (not eq, not in) match
+   * return {@code Float.NaN} for not applicable
+   */
+  default float getNumMatchingItems() {
+    return Float.NaN;
+  };
```

Comment:
Why returning float here? You may return `int` and use `Integer.MIN_VALUE` for non-applicable
```suggestion
  default int getNumMatchingValues() {
    return Integer.MIN_VALUE;
  };
```

Reply:
Done